lib/parserm/parserM.mli

```diff
@@ -0,0 +1,28 @@
+module type Distinguishable = sig
+    type t
+    val equal : t -> t -> bool  
+end
+
+module type ParserMSig = sig
+    (** [ParserM] is a signature for the parser monad
+        @author Raaid Tanveer **)
+
+    (** [parser_m] is the type for the parser monad that takes in a token list and 
+        produces an abstract syntax tree of parametrized type 'a **)
+    type token
+    type 'a parser_m = token list -> ('a * token list, string) result 
```

Comment:
This type would ideally be abstract, with an additional function called `parse : 'a parser_m -> token list -> ('a * token list, string) result)`.